wearipedia/devices/biostrap/evo_gen.py

```diff
@@ -0,0 +1,248 @@
+import random
+from datetime import datetime, timedelta
+
+
+def create_syn_data(start_date, end_date):
+    """
+    Generates synthetic data collected by Biostrap between a given start and end date.
+
+    :param start_date: Start date (inclusive) as a string in the format "YYYY-MM-DD"
+    :type start_date: str
+    :param end_date: End date (inclusive) as a string in the format "YYYY-MM-DD"
+    :type end_date: str
+
+    :return: A tuple consisting of:
+        - activities: Dictionary containing details of a random synthetic activity
+        - bpm: Dictionary representing beats per minute for every 10 seconds throughout the range
+        - brpm: Dictionary representing breaths per minute based on bpm values
+        - hrv: Dictionary representing heart rate variability for every 10 seconds throughout the range
+        - resting_bpm: Dictionary representing resting beats per minute for every 10 seconds
+        - resting_hrv: Dictionary representing resting heart rate variability for every 10 seconds
+        - spo2: Dictionary representing blood oxygen saturation for every 10 seconds
+        - rest_cals: Dictionary representing resting calories burned each day
+        - work_cals: Dictionary representing work-related calories burned each day
+        - active_cals: Dictionary representing active calories burned each day
+        - step_cals: Dictionary representing calories burned from steps each day
+        - total_cals: Dictionary representing total calories burned each day
+        - sleep_session: Dictionary representing moments of movement during typical sleeping hours
+        - sleep_detail: Dictionary representing details of a synthetic sleep session
+        - steps: Dictionary representing steps taken every minute throughout the range
+        - distance: Dictionary representing distance covered (based on steps) every minute throughout the range
+
+    :rtype: Tuple[Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict, Dict]
+    """
+
+    # Convert the provided strings to datetime objects
+    start_date_obj = datetime.strptime(start_date, "%Y-%m-%d")
+    end_date_obj = datetime.strptime(end_date, "%Y-%m-%d")
+
+    TZ_OFFSET = -420
+
+    # Helper functions for biometric data generation
+    def synthetic_biometrics(start_date_obj, end_date_obj):
```

Comment:
It looks like here your high frequency data is uniformly random. I think a good improvement here would be to use a more complex generator to make it look more realistic. Some example generators exist in Garmin Fenix 7s, Polar Verity Sense

Reply:
It looks like the data is still pretty noisy, could you try another generator?